src/QueryPipeline/RemoteQueryExecutorReadContext.h

```diff
@@ -44,11 +44,11 @@ class RemoteQueryExecutorReadContext : public AsyncTaskExecutor
 
     Packet getPacket();
 
-    UInt64 getPacketType();
+    UInt64 getPacketType() const;
 
-    bool hasReadPacketType() const { return has_read_packet_type; }
+    bool hasReadPacketType() const { return has_read_packet_part == PacketPart::Type; }
```

Comment:
```suggestion
    bool hasReadPacketType() const { return has_read_packet_part >= PacketPart::Type; }
```

?

Reply:
Logically fit the method name, but I'd not relax the condition, so let's rename the method.